src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -826,8 +826,17 @@ bool clang_cpp_convertert::get_constructor_call(
    */
 
   // Calling base constructor from derived constructor
+  bool need_new_obj = false;
+
   if (new_expr.base_ctor_derived())
     gen_typecast_base_ctor_call(callee_decl, call, new_expr);
+  else if (new_expr.get_bool("#member_init"))
+  {
+    /*
+    struct A {}; struct B {A a; B(int _a) : a(_a) {}};
+    In this case, use variables to construct the object directly
```

Comment:
`A` needs a constructor taking an int. Also, I'd replace "variables" by "members" for clarity.